.github/workflows/tests.yml

```diff
@@ -320,3 +320,13 @@ jobs:
       - run: brew test-bot --only-formulae --test-default-formula
 
       - uses: codecov/codecov-action@29386c70ef20e286228c72b668a06fd0e8399192
+
+      - name: Upload test results to BuildPulse for flaky test detection
+        if: '!cancelled()' # Run this step even when the tests fail. Skip if the workflow is cancelled.
```

Comment:
```suggestion
        # Only run this step for PRs where where we have access to secrets.
        # Run this step even when the tests fail. Skip if the workflow is cancelled.
        if: github.event.pull_request.head.repo.full_name == github.repository && !cancelled()
```

Reply:
It looks like this conditional prevented this step also from running on the `master` branch (https://github.com/Homebrew/brew/runs/2893711700?check_suite_focus=true), presumably because there was no pull request present for that build. I think we want this step to always run for builds on the master branch, so I'll open a pull request to adjust this conditional.

Reply:
That'd be great, thanks @jasonrudolph!

Reply:
Cool. I pushed up https://github.com/Homebrew/brew/pull/11586 with the proposed change.